tests/db_functions/tests.py

```diff
@@ -547,3 +547,74 @@ def test_now(self):
             ['How to Time Travel'],
             lambda a: a.title
         )
+
+    def test_length_transform(self):
+        try:
+            CharField.register_transform(Length, 'length')
```

Comment:
I've moved register_lookup context manager to django.db.models.query_utils, and created register_transform to go with it. I haven't used it in these tests though, because these tests use the longer form of registration that provides a lookup_name. That's incompatible with the `register_lookup(field, *lookups)` API.